### PR TITLE
[VM] Allow two VM instance create in one thread

### PR DESCRIPTION
**Pull request by ibmibmibm**
* Add test for multiple VM instance

### CODE REVIEW

File: include/vm/async.h [hunk 1]
```diff
@@ -34,7 +34,6 @@ template <typename T> class Async {
         std::thread([FPtr, P = std::move(Promise),
                      Tuple = std::tuple(
                          &TargetVM, std::forward<ArgsT>(Args)...)]() mutable {
-          std::get<0>(Tuple)->newThread();
```
Comment (hydai) [inline include/vm/async.h:37]: Do we need to call prepare here?
It looks like the `newThread` is replaced by `prepare`.

**Review by hydai — CHANGES_REQUESTED** [review]

File: include/vm/async.h (hunk 1 above)
Comment (ibmibmibm) [reply include/vm/async.h:37]: No more `prepare` required here.

**Review by hydai — DISMISSED** [review]
LGTM

**Review by hydai — APPROVED** [review]

**Comment by juntao** [comment]
Flows review

**Pull request merged**